drivers/i2s/i2s_ll_stm32.c

```diff
@@ -175,6 +180,12 @@ static int i2s_stm32_set_clock(const struct device *dev,
 	i2s_odd = (i2s_div & 0x1) ? 1 : 0;
 	i2s_div >>= 1;
 
+	/* i2s_div == 0 || i2s_div == 1 are forbidden */
+	if (i2s_div < 2U) {
```

Comment:
In case,  i2s_div < 2U, the LL_RCC_SetI2SClockSource is useless, consequently checking i2s_div and exiting in case of error could be earlier in the function : 
```suggestion
	if (i2s_div < 2U) {
		LOG_ERR("The linear prescaler value is unsupported");
		return -ENOTSUP;
	}

	/* Select clock source */
	LL_RCC_SetI2SClockSource(cfg->i2s_clk_sel);

```

Reply:
That makes perfect sense, thank you.
I will also apply the same train of thought to `LL_I2S_Enable/DisableMasterClock`